### PR TITLE
Extend base Trainer with LoRATrainer using PEFT

### PR DESCRIPTION
**Pull request by SaviNimz**
This PR introduces a LoRATrainer class that extends the base Trainer to support Parameter-Efficient Fine-Tuning (PEFT) using LoRA via the peft library. If a LoraConfig is provided, the model is wrapped using get_peft_model; otherwise, it behaves exactly like the standard Trainer.

### CODE REVIEW

**Review by blkdmr — CHANGES_REQUESTED** [review]
Hi, the code is well written, but there are a few issues that make this class unusable as it is. For example, at least a base LLM model, a tokenizer and a PEFT config object should be passed to the trainer. Also, the training procedure needs to be reworked, since we have attention masks, tokenizers, and other components that aren’t supported by the base Trainer class. Do you think you could address these issues?
